SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -21,62 +22,95 @@ SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName&
+{
+	// Should only be passed auth intent or ACL.
+	check(Op.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID || Op.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID);
+
+	if (AclAssignmentRequestIsQueued(Op.entity_id))
 	{
-		QueueAclAssignmentRequest(Op.entity_id);
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Log,
+			TEXT("Component %d for entity %lld removed. Can no longer enforce the previous request for this entity."),
+			Op.component_id, Op.entity_id);
+		AclWriteAuthAssignmentRequests.Remove(Op.entity_id);
 	}
 }
 
-// This is called whenever this worker becomes authoritative for the ACL component on an entity.
-// It is now this worker's responsibility to make sure that the ACL reflects the current intent,
-// which may have been received before this call.
-void SpatialLoadBalanceEnforcer::AuthorityChanged(const Worker_AuthorityChangeOp& AuthOp)
+void SpatialLoadBalanceEnforcer::OnEntityRemoved(const Worker_RemoveEntityOp& Op)
 {
-	if (AuthOp.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID &&
-		AuthOp.authority == WORKER_AUTHORITY_AUTHORITATIVE)
+	if (AclAssignmentRequestIsQueued(Op.entity_id))
 	{
-		const SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(AuthOp.entity_id);
-		if (AuthorityIntentComponent == nullptr)
-		{
-			// TODO(zoning): There are still some entities being created without an authority intent component.
-			// For example, the Unreal created worker entities don't have one. Even though those won't be able to
-			// transition, we should have the intent component on them for completeness.
-		 	UE_LOG(LogSpatialLoadBalanceEnforcer, Log, TEXT("Requested authority change for entity without AuthorityIntent component. EntityId: %lld"), AuthOp.entity_id);
-			return;
-		}
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Log, TEXT("Entity %lld removed. Can no longer enforce the previous request for this entity."),
+			Op.entity_id);
+		AclWriteAuthAssignmentRequests.Remove(Op.entity_id);
+	}
+}
 
-		const PhysicalWorkerName* OwningWorkerId = VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(AuthorityIntentComponent->VirtualWorkerId);
-		if (OwningWorkerId != nullptr &&
-			*OwningWorkerId == WorkerId &&
-			StaticComponentView->HasAuthority(AuthOp.entity_id, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Verbose, TEXT("No need to queue newly authoritative entity %lld because this worker is already authoritative."), AuthOp.entity_id);
-			return;
-		}
-		QueueAclAssignmentRequest(AuthOp.entity_id);
+void SpatialLoadBalanceEnforcer::OnAclAuthorityChanged(const Worker_AuthorityChangeOp& AuthOp)
+{
+	// This class should only be informed of ACL authority changes.
+	check(AuthOp.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID);
+
+	if (AclAssignmentRequestIsQueued(AuthOp.entity_id) && AuthOp.authority != WORKER_AUTHORITY_AUTHORITATIVE)
```

Comment:
```suggestion
	if (AuthOp.authority != WORKER_AUTHORITY_AUTHORITATIVE && AclAssignmentRequestIsQueued(AuthOp.entity_id))
```

Reply:
done

Reply:
actually changed them to separate if statements